src/controllers/queryRunner.ts

```diff
@@ -137,30 +144,32 @@ export default class QueryRunner {
                 };
 
                 // Send the request to execute the query
-                return this._client.sendRequest(QueryExecuteStatementRequest.type, queryDetails).then(onSuccess, onError);
+                await this._client.sendRequest(QueryExecuteStatementRequest.type, queryDetails).then(onSuccess, onError);
             });
     }
 
     // Pulls the query text from the current document/selection and initiates the query
-    public runQuery(selection: ISelectionData): Thenable<void> {
-        return this.doRunQuery(
+    public async runQuery(selection: ISelectionData, promise?: Deferred<boolean>): Promise<void> {
+        await this.doRunQuery(
             selection,
-            (onSuccess, onError) => {
+            async (onSuccess, onError) => {
                // Put together the request
                 let queryDetails: QueryExecuteParams = {
                     ownerUri: this._ownerUri,
                     querySelection: selection
                 };
 
                 // Send the request to execute the query
-                return this._client.sendRequest(QueryExecuteRequest.type, queryDetails).then(onSuccess, onError);
+                if (promise) {
+                    this._uriToQueryPromiseMap.set(this._ownerUri, promise);
+                }
+                await this._client.sendRequest(QueryExecuteRequest.type, queryDetails).then(onSuccess, onError);
             });
     }
 
     // Pulls the query text from the current document/selection and initiates the query
-    private doRunQuery(selection: ISelectionData, queryCallback: any): Thenable<void> {
-        const self = this;
-        self._vscodeWrapper.logToOutputChannel(Utils.formatString(LocalizedConstants.msgStartedExecute, this._ownerUri));
+    private async doRunQuery(selection: ISelectionData, queryCallback: any): Promise<void> {
+        this._vscodeWrapper.logToOutputChannel(Utils.formatString(LocalizedConstants.msgStartedExecute, this._ownerUri));
```

Comment:
nit - I know it was like this before but generally it's a good idea to type callbacks so you're sure that the function signature is as expected. So it'd look like this `queryCallback: (onSuccess: (result) => void, onError: (reason) => void) => void` (you might even be able to get the types for result and reason based on what sendRequest returns)

Reply:
Will do it as a part of the cleanup.